test/uk/gov/hmrc/customs/emailfrontend/connectors/Save4LaterConnectorSpec.scala

```diff
@@ -16,23 +16,27 @@
 
 package uk.gov.hmrc.customs.emailfrontend.connectors
 
-import play.api.http.Status._
+import org.mockito.Mockito.when
+import play.api.http.Status.*
 import play.api.libs.json.Format.GenericFormat
 import play.api.libs.json.OFormat.oFormatFromReadsAndOWrites
-import play.api.libs.json.{JsValue, Json}
+import play.api.libs.json.Json
 import play.api.test.Helpers.running
 import play.api.{Application, inject}
 import uk.gov.hmrc.customs.emailfrontend.connectors.http.responses.{BadRequest, UnhandledException}
 import uk.gov.hmrc.customs.emailfrontend.model.{EmailDetails, JourneyType, ReferrerName}
 import uk.gov.hmrc.customs.emailfrontend.utils.{FakeIdentifierAgentAction, SpecBase}
-import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier, HttpClient, HttpResponse, SessionId}
+import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier, HttpReads, HttpResponse, SessionId}
 import uk.gov.hmrc.customs.emailfrontend.utils.Utils.emptyString
+import uk.gov.hmrc.http.client.{HttpClientV2, RequestBuilder}
+import org.mockito.ArgumentMatchers.any
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 class Save4LaterConnectorSpec extends SpecBase {
 
-  val mockHttpClient = mock[HttpClient]
+  val mockHttpClient = mock[HttpClientV2]
```

Comment:
Better to add the return type for this and for line no 39 as these are public members